src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts

```diff
@@ -141,6 +158,53 @@ class TerminalSuggestContribution extends DisposableStore implements ITerminalCo
 		}
 	}
 
+	// TODO: Eventually support multiple LSP providers for non-Python REPLs.
+	private async _loadLspCompletionAddon(xterm: RawXtermTerminal): Promise<void> {
+		const isWsl =
+			isLinux &&
+			(
+				!!env['WSL_DISTRO_NAME'] ||
+				!!env['WSL_INTEROP']
+			);
+
+		// Windows, WSL currently does not support shell integration for Python REPL.
+		if (isWindows || isWsl) {
+			return;
+		}
+
+		if (
+			this._ctx.instance.shellType !== GeneralShellType.Python &&
```

Comment:
could we create consts for all of these strings and keep them in lsp files? `python`, `py`, and `ms-python...`

Reply:
Great idea, just did them here: [58e810f](https://github.com/microsoft/vscode/pull/243528/commits/58e810f92eb8ab99bece078b254ce12f961a11b8)

Also I removed check for `this._ctx.instance.shellLaunchConfig.executable` since it seems like its never Python but always upper shell. `this._ctx.shellType` would be enough check.